pkg/client/restclient/request.go

```diff
@@ -821,9 +821,16 @@ func (r *Request) request(fn func(*http.Request, *http.Response)) error {
 		}
 
 		done := func() bool {
-			// ensure the response body is closed before we reconnect, so that we reuse the same
-			// TCP connection
-			defer resp.Body.Close()
+			// Ensure the response body is fully read and closed
+			// before we reconnect, so that we reuse the same TCP
+			// connection.
+			defer func() {
+				const maxBodySlurpSize = 2 << 10
```

Comment:
You're right though... the -1 should not need to be explicitly checked since it satisfies the other condition. I just cribbed this from the standard lib.

Body should never be nil. Even if it's empty. The http package explicitly guarantees this.